Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx

```diff
@@ -1,44 +1,108 @@
 import Heading, { HeadingType } from '../guidelines/headings/Heading';
+import SpinningLoader from '../guidelines/spinning-loader/SpinningLoader';
 
 import styles from './LoginForm.module.scss';
 
 const LoginPage = () => {
     const {
-        state: { loginErrorMessage },
+        state: {
+            loginErrorMessage,
+            isLoggingIn,
+            isLoggedIn,
+        },
         actions: {
             setUsername,
             setPassword,
             signIn,
+            setLoginErrorMessage,
         },
     } = useAuth();
+
+    const [ usernameFormError, setUsernameFormError ] = useState('');
+    const [ passwordFormError, setPasswordFormError ] = useState('');
+    const [ disableLoginButton, setDisableLoginButton ] = useState(false);
+    const [ hasPressedLoginBtn, setHasPressedLoginBtn ] = useState(false);
+
     const usernameFieldName = 'Username';
     const passwordFieldName = 'Password';
 
     const handleOnChangeUpdateUsername = useCallback((value?: IFormControlOnChangeValueType) => {
+        if (isEmpty(value)) {
+            setUsernameFormError(EmptyUsernameErrorMessage);
+        } else if (!isNil(value) && (value.length < 5 || value.length > 32)) {
+            setUsernameFormError(UsernameLengthErrorMessage);
+        } else {
+            const regex = /^[a-zA-Z][a-zA-Z0-9._]{3,30}[a-zA-Z0-9]$/;
+            if (!regex.test(value as string)) {
+                setUsernameFormError(UsernameFormatErrorMessage);
+            } else {
+                setUsernameFormError('');
+            }
+        }
+
         setUsername(isNil(value)
             ? ''
             : value as string);
     }, [ setUsername ]);
 
     const handleOnChangeUpdatePassword = useCallback((value?: IFormControlOnChangeValueType) => {
+        if (isEmpty(value)) {
+            setPasswordFormError(EmptyPasswordErrorMessage);
+        } else if (!isNil(value) && value.length < 6) {
+            setPasswordFormError(PasswordLengthErrorMessage);
+        } else {
+            setPasswordFormError('');
+        }
+
         setPassword(isNil(value)
             ? ''
             : value as string);
     }, [ setPassword ]);
 
+    useEffect(() => {
+        if (!isEmpty(usernameFormError) && hasPressedLoginBtn) {
+            setLoginErrorMessage(usernameFormError);
+            setDisableLoginButton(true);
+            return;
+        }
+
+        if (!isEmpty(passwordFormError) && hasPressedLoginBtn) {
+            setLoginErrorMessage(passwordFormError);
+            setDisableLoginButton(true);
+            return;
+        }
+
+        setLoginErrorMessage('');
+        setDisableLoginButton(false);
+    }, [ usernameFormError, passwordFormError, setLoginErrorMessage, hasPressedLoginBtn ]);
+
     const handleLoginClick = useCallback(async () => {
         /* TODO:  Add message to notify the admin if SULS is not working.
```

Comment:
Is this implemented?

Reply:
Not implemented. Will be implemented when Redux is introduced to this page.